offchain/inspect-server/src/config.rs

```diff
@@ -38,30 +44,29 @@ pub struct Config {
 impl Config {
     /// Generate config from command line arguments and and environment variables.
     /// If config path is provided, read configuration from the file.
-    pub fn initialize() -> Result<Self, BadConfigurationError> {
+    pub fn initialize() -> Result<Self, ConfigError> {
         let env_cli_config = EnvCLIConfig::from_args();
         let file_config: FileConfig =
-            load_config_file(env_cli_config.config_path)
-                .map_err(|e| BadConfigurationError { err: e.to_string() })?;
+            load_config_file(env_cli_config.config_path)?;
 
         let inspect_server_address: String = env_cli_config
             .inspect_server_address
             .or(file_config.inspect_server_address)
-            .ok_or(BadConfigurationError {
+            .context(BadConfigurationSnafu {
                 err: String::from("Must specify inspect server address"),
```

Comment:
I think `"Must specify inspect server address".to_string()` is prettier. Hehehe.